cmod/conditions.py

```diff
@@ -0,0 +1,160 @@
+import json
+import datetime
+import os
+
+# NOTE: "uses visual system commands"
+class Conditions(object):
+  def __init__(self, cmd):
+    self.cmd = cmd
+    self.logger = cmd.devlog("Conditions")
+    # gantry conditions should be stored as a dictionary
+    self.gantry_conditions = {}
+    self.gantry_conditions_use_count = 0
+    self.gantry_conditions_filename = ""
+    self.h_list = []
+
+  # loads gantry conditions from a file and returns True if successful, False otherwise
+  def load_gantry_conditions(self, file):
+    self.gantry_conditions_filename = file
+    conditions = json.loads(open(file, 'r').read())
+    try:
+      self.gantry_conditions = {
+        "FOV_to_gantry_coordinates": {
+          "z": conditions["FOV_to_gantry_coordinates"]["z"],
+          "transform": conditions["FOV_to_gantry_coordinates"]["data"]["transform"]
+        },
+        "lumi_vs_FOV_center": {
+          "z": conditions["lumi_vs_FOV_center"]["z"],
+          "separation": conditions["lumi_vs_FOV_center"]["data"]["separation"]
+        },
+        "use_count": conditions["use_count"] if "use_count" in conditions else 0
+      }
+
+      self.increment_use_count()
+
+      self.h_list = [self.gantry_conditions["lumi_vs_FOV_center"]["data"]["separation"]]
+
+      return True
+    except KeyError as e:
+      self.logger.error(e)
+      self.logger.error("""
+        The gantry conditions file does not contain the required gantry conditions:
+        'FOV_to_gantry_coordinates', and 'lumi_vs_FOV_center'. Please check the
+        file and the required format and try again.""")
+      # TODO: might want to add logic that if the required conditions are not provided, run the process to calculate those that are missing or even all of them as don't want to trust an "incomplete" set of conditions
+      return False
+
+  # saves gantry conditions to a file
+  def save_gantry_conditions(self, filename=""):
+    if filename != "":
+      self.gantry_conditions_filename = filename
+    if self.gantry_conditions_filename == "":
+      self.create_gantry_conditions_filename()
+      self.increment_use_count()
+
+    with open(self.gantry_conditions_filename, 'w') as f:
+      f.write(json.dumps(self.gantry_conditions, indent=2))
+  
+  # returns the gantry conditions
+  def get_gantry_conditions(self):
+    return self.gantry_conditions
+  
+  def update_gantry_and_sipm_conditions(self, cmd, detid, z):
```

Comment:
conflicting name `cmd` and `self.cmd` (the command line session object). While we can have `self.cmd` and `cmd` does mean different objects, I think we should move the function argument here to be `cmdname`.

Reply:
Agree. Good point.